dotnet/samples/KernelSyntaxExamples/Example78_LMStudio.cs

```diff
@@ -0,0 +1,132 @@
+    {
+        // Setup Llama2 as the model in LM Studio UI.
+
+        var kernel = Kernel.CreateBuilder()
+            .AddLMStudioChatCompletion()
+            .Build();
+
+        var prompt = @"Rewrite the text between triple backticks into a business mail. Use a professional tone, be clear and concise.
+                   Sign the mail as AI Assistant.
+
+                   Text: ```{{$input}}```";
+
+        var mailFunction = kernel.CreateFunctionFromPrompt(prompt, new OpenAIPromptExecutionSettings
+        {
+            Temperature = 0.7,
+            MaxTokens = 1000,
+        });
+
+        var response = await kernel.InvokeAsync(mailFunction, new() { ["input"] = "Tell David that I'm going to finish the business plan by the end of the week." });
+        this.WriteLine(response);
+    }
+
+    [Fact(Skip = "Manual configuration needed")]
+    public async Task Llama2StreamingExampleAsync()
+    {
+        // Setup Llama2 as the model in LM Studio UI.
+
+        var kernel = Kernel.CreateBuilder()
+            .AddLMStudioChatCompletion()
+            .Build();
+
+        var prompt = @"Rewrite the text between triple backticks into a business mail. Use a professional tone, be clear and concise.
+                   Sign the mail as AI Assistant.
+
+                   Text: ```{{$input}}```";
+
+        var mailFunction = kernel.CreateFunctionFromPrompt(prompt, new OpenAIPromptExecutionSettings
+        {
+            Temperature = 0.7,
+            MaxTokens = 1000,
+        });
+
+        await foreach (var word in kernel.InvokeStreamingAsync(mailFunction, new() { ["input"] = "Tell David that I'm going to finish the business plan by the end of the week." }))
+        {
+            this.WriteLine(word);
+        };
+    }
+
+    [Fact(Skip = "Manual configuration needed")]
+    public async Task Phi2StreamingExampleAsync()
+    {
+        var kernel = Kernel.CreateBuilder()
+            .AddLMStudioChatCompletion()
+            .Build();
+
+        string prompt = "Write a short poem about cats";
+
+        await foreach (string word in kernel.InvokePromptStreamingAsync<string>(prompt))
+        {
+            this.Write(word);
+        };
+    }
+
+    public Example78_LMStudio(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    public sealed class MyHttpMessageHandler : HttpClientHandler
+    {
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request.RequestUri != null && request.RequestUri.Host.Equals("api.openai.com", StringComparison.OrdinalIgnoreCase))
+            {
+                request.RequestUri = new Uri($"http://localhost:1234{request.RequestUri.PathAndQuery}");
+            }
+
+            return base.SendAsync(request, cancellationToken);
+        }
+    }
+}
+
+public static class KernelBuilderExtensions
+{
+    public static IKernelBuilder AddLMStudioChatCompletion(this IKernelBuilder builder)
```

Comment:
Can we create an extension method in the OpenAI connector project with a signature something like this:

`public static IKernelBuilder AddCustomChatCompletion(this IKernelBuilder builder, Uri requestUri)`

And then have samples that use this with LM Studio and the new Ollama. This would give people an OOB option to use our OpenAI connector with local model that are OpenAI REST API compatible.

Reply:
> AddCustomChatCompletion

I'd personally prefer either AddOpenAICompatibleChatCompletion or just adding overloads of the existing OpenAI methods to also take a uri. Do we know what Azure.AI.OpenAI's plans are for also making this extensible?

If we do add such methods, some care will need to be taken to make them work well with DI, given that the implementation is based on supplying a custom message handler.

Reply:
Please also create same extension for `IServiceCollection`